client/scripts/views/components/new_thread_form.ts

```diff
@@ -165,12 +166,15 @@ export const NewThreadForm: m.Component<{
     vnode.state.recentlyDeletedDrafts = [];
     vnode.state.uploadsInProgress = 0;
     vnode.state.overwriteConfirmationModal = false;
+    if (app.community?.meta.topics.length || app.chain?.meta.topics.length) {
+      vnode.state.hasTopics = true;
```

Comment:
Why are we assigning this to `vnode.state`? We should avoid saving state in components unless necessary.

Reply:
Well it gets used in the `oninit` and the `view` lifecycles; that requires state, yeah?

Reply:
It never gets reassigned though - shouldn't we use vnode.attrs directly?

Reply:
Sure, lemme check that out.